### PR TITLE
Add PR template to repository

### PR DESCRIPTION
**Pull request by hCraker**
This PR is part of the preparation for interns this summer and will also benefit outside contributors. A PR template has been added with a check list for function development/testing and documentation. Please let me know if you think anything should be changed. I'd like to add a similar template to our other repositories if this one is helpful.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  issue #229".
+
+- The summary should provide at least 1-2 sentences describing the pull request
+  in detail (Why is this change required?  What problem does it solve?) and
+  link to any relevant issues.
+
+If you need assistance with your PR, please let the GeoCAT team know by
+tagging us with @geocat. We can help if reviews are unclear, the recommended changes
+seem overly demanding, you would like help in addressing a reviewer's comments,
+or if you have been waiting more than a week to hear back on your PR.
```
Comment (hCraker) [inline .github/PULL_REQUEST_TEMPLATE.md:46]: If you all think there is a better way to contact us (such as the email) let me know. I also want your opinions on if waiting a week for a response to a PR is too long. We generally move quickly, but I'd hate for an outside contributor to get frustrated if we are swamped with something else and take a few days to reply/review.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+- [ ] Docstrings have updated with any function changes
+- [ ] Internal functions have a preceeding underscore (`_`) and have been added to `docs/internal_api/index.rst`
+- [ ] User facing functions have been added to `docs/user_api/index.rst` under their module
+- [ ] Appropriate labels have been added to this PR
+
+**PR Etiquette Reminders**
+- This PR should be listed as a draft PR until you are ready to request reviewers
+- After making changes in accordance with the reviews, re-request your reviewers
+- Do *not* mark conversations as resolved if you didn't start them
+- Do mark conversations as resolved *if you opened them* and are satified with the changes/discussion.
```
Comment (anissa111) [inline .github/PULL_REQUEST_TEMPLATE.md:24]: Oh I have absolutely been resolving conversations when I fix them if they're a small enough thing, whoops sorry!

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+- [ ] tests for function exists in associated module test file
+- [ ] tests cover all possible logical paths in your function
+
+**Documentation**
+- [ ] Passes `precommit`. If it fails, try running `pre-commit run --all` and push changes
+- [ ] Docstrings have been added to all new functions ([Documentation Standards](https://geocat.ucar.edu/pages/contributing.html#422-documentation))
+- [ ] Docstrings have updated with any function changes
+- [ ] Internal functions have a preceeding underscore (`_`) and have been added to `docs/internal_api/index.rst`
+- [ ] User facing functions have been added to `docs/user_api/index.rst` under their module
+- [ ] Appropriate labels have been added to this PR
```
Comment (anissa111) [inline .github/PULL_REQUEST_TEMPLATE.md:18]:
```suggestion
## PR Checklist
<!-- Please mark any checkboxes that do not apply to this PR as [N/A]. -->
**Functionality**
- [ ] Function is in appropriate module file
- [ ] New function(s) intended for public API added to `src/geocat/comp/__init__.py` file

**Testing**
- [ ] Tests for function exists in associated module test file
- [ ] Tests cover all possible logical paths in your function

**Documentation**
- [ ] Passes `precommit`. To set up on your local, run `pre-commit install` from the top level of the repository. To manually run pre-commits, use `pre-commit run --all-files` and re-add any changed files before committing again and pushing.
- [ ] Docstrings have been added to all new functions ([Documentation Standards](https://geocat.ucar.edu/pages/contributing.html#422-documentation))
- [ ] Docstrings have updated with any function changes
- [ ] Internal functions have a preceeding underscore (`_`) and have been added to `docs/internal_api/index.rst`
- [ ] User facing functions have been added to `docs/user_api/index.rst` under their module
- [ ] Appropriate labels have been added to this PR

**Examples**
- [ ] Any new notebook examples added to `docs/examples/` folder
- [ ] Pre-run all notebook cells
- [ ] New notebook files added to `docs/examples.rst` toctree
- [ ] New notebook files added to new entry in `docs/gallery.yml` with appropriate thumbnail photo in `docs/_static/thumbnails/`
```

Comment (anissa111) [inline .github/PULL_REQUEST_TEMPLATE.md:18]: I'm not sure this point should go in **Documentation**. Maybe we should add a **General** section with like
- tags
- use a fork
- mark as draft if not ready for input
- take out of draft before requesting reviewers

Stuff that's a little more concrete than general etiquette, but is generalized to all PRs? Stuff that can be checked off, you know?

Comment (anissa111) [reply .github/PULL_REQUEST_TEMPLATE.md:18]: Sorry that's a huge diff. I moved a couple of points around and added an **Examples** and **Functionality** section. Definitely feel free to disagree with me on any of this, but I did want to include some instructions specifically for examples PRs

**Review by anissa111 — COMMENTED** [review]
Gosh, we really have needed this, thanks for taking a pass at it! I've left some conversations below.

**Comment by anissa111** [comment]
Oh, also, should we provide instructions on who to request for reviews? We could list a couple of us specifically, but I think requesting the team as a whole might work better. Thoughts?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (hCraker) [reply .github/PULL_REQUEST_TEMPLATE.md:24]: This is general GitHub etiquette and something I was told to follow when I first was hired. You and I rarely resolve things prematurely (same with everyone on the team honestly), but I wanted to add it to the template for external folks.

[the remark this answers is not in the document]
Comment (hCraker) [reply .github/PULL_REQUEST_TEMPLATE.md:24]: Yeah good idea. I wasn't sure if this though be a comment or not. I was worried comments wouldn't be always be read

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+## PR Summary
+<!-- Summary goes here. Replace XXX with the number of the issue this PR will resolve. -->
```
[the remark this answers is not in the document]
Comment (hCraker) [reply .github/PULL_REQUEST_TEMPLATE.md:2]: Ah yes! I forgot to add that. Good catch!

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (hCraker) [reply .github/PULL_REQUEST_TEMPLATE.md:21]: Yes let's do that. I wasn't sure the best way to organize all of the info, but a general section is a great idea

**Comment by hCraker** [comment]
> Oh, also, should we provide instructions on who to request for reviews? We could list a couple of us specifically, but I think requesting the team as a whole might work better. Thoughts?

I think requesting the whole team isn't a bad idea. It's more robust if we have staffing changes, and it keeps everyone in the know about outside contributions. The only downside is non-comp staff getting emails they don't need/want, but they can unsub from the PR as they please

**Review by anissa111 — APPROVED** [review]
Yeah this looks good! I think we'll probably want to iterate on this when we test it out with, like, interns, but I can't think of anything else that we should put here.

**Pull request merged**